pkg/chain/ethereum/ethereum.go

```diff
@@ -318,6 +319,144 @@ func (bc *baseChain) OperatorKeyPair() (
 	return privateKey, publicKey, nil
 }
 
+// GetBlockNumberByTimestamp gets the block number for the given timestamp.
+// In the best case, the block with the exact same timestamp is returned.
+// If the aforementioned is not possible, it tries to return the closest
+// possible block. However, there may be cases where another close block
+// is returned instead of the closest one. This may happen if the actual
```

Comment:
>  However, there may be cases where another close block is returned instead of the closest one.

Can you give an example? Should the two final loops `for block.Time() < timestamp` and `if block.Time() > timestamp` fix it?

Reply:
Yeah, this is no longer the case. Fixed this as part of 9e52fa1f